ksqldb-functional-tests/src/test/resources/query-validation-tests/binary-comparison.json

```diff
@@ -182,37 +181,53 @@
       ]
     },
     {
-      "name": "comparison array fails",
+      "name": "array comparison",
```

Comment:
can you add tests for nested data types? I'm not entirely sure what happens in that case (e.g. an array of structs, or a struct with a nested struct/array)

Reply:
Likewise, add similar style tests covering `IS DISTINCT FROM` and `IS NOT DISTINCT FROM`.